### PR TITLE
[X86] Support -march=pantherlake,clearwaterforest

### PR DESCRIPTION
**Pull request by FreddyLeaf**

### CODE REVIEW

File: clang/test/Preprocessor/predefined-arch-macros.c [hunk 1]
```diff
@@ -2628,6 +2647,9 @@
 // CHECK_SRF_M64: #define __PCONFIG__ 1
 // CHECK_SRF_M64: #define __PKU__ 1
 // CHECK_SRF_M64: #define __POPCNT__ 1
+// CHECK_SRF_M64-NOT: #define __PREFETCHI__ 1
+// CHECK_ARLS_M64-NOT: #define __PREFETCHI__ 1
+// CHECK_PTL_M64: #define __PREFETCHI__ 1
```
Comment (phoebewang) [inline clang/test/Preprocessor/predefined-arch-macros.c:2652]: Missing `CHECK_CWF_M64`?

File: clang/test/Preprocessor/predefined-arch-macros.c [hunk 2]
```diff
@@ -2538,6 +2544,9 @@
 // CHECK_SRF_M32: #define __PCONFIG__ 1
 // CHECK_SRF_M32: #define __PKU__ 1
 // CHECK_SRF_M32: #define __POPCNT__ 1
+// CHECK_SRF_M32-NOT: #define __PREFETCHI__ 1
+// CHECK_ARLS_M32-NOT: #define __PREFETCHI__ 1
+// CHECK_PTL_M32: #define __PREFETCHI__ 1
```
Comment (phoebewang) [inline clang/test/Preprocessor/predefined-arch-macros.c:2549]: Missing `CHECK_CWF_M64`?

File: llvm/lib/Target/X86/X86.td [hunk 3]
```diff
@@ -1249,6 +1249,18 @@ def ProcessorFeatures {
   list<SubtargetFeature> ARLSFeatures =
     !listconcat(SRFFeatures, ARLSAdditionalFeatures);
 
+  // Pantherlake
+  list<SubtargetFeature> PTLAdditionalFeatures = [FeaturePREFETCHI];
+  list<SubtargetFeature> PTLFeatures =
+    !listconcat(ARLSFeatures, PTLAdditionalFeatures);
```
Comment (phoebewang) [inline llvm/lib/Target/X86/X86.td:1255]: EAS doesn't say `Panther Lake` has these `ARLSAdditionalFeatures`

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86.td:1255]: I referred to xed patch: https://github.com/intelxed/xed/blob/main/datafiles/ptl/ptl-chips.txt

File: clang/test/Preprocessor/predefined-arch-macros.c (hunk 1 above)
Comment (FreddyLeaf) [reply clang/test/Preprocessor/predefined-arch-macros.c:2652]: same above

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**